ruby/src/main/java/com/google/protobuf/jruby/RubyMessage.java

```diff
@@ -411,6 +410,7 @@ private RubyRepeatedField getRepeatedField(ThreadContext context, Descriptors.Fi
         for (int i = 0; i < count; i++) {
             ret.push(context, wrapField(context, fieldDescriptor, this.builder.getRepeatedField(fieldDescriptor, i)));
         }
+        addRepeatedField(fieldDescriptor, ret);
```

Comment:
if `getRepeatedField` is called, and the repeatedField has not previously been instantiated, the method was instantiating one, but not adding it to itself.  This means that methods like `<<` and `push` were creating a new repeatedField, but it wasn't being saved to the parent object.  Only methods such as `+=` worked because of the `=`, which called `setField`, which in turn called `addRepeatedField`.  See the added tests, which will fail incorrectly if run on the `master` branch.